test/integration.sh

```diff
@@ -7,41 +7,75 @@
 #       example: bash integration.sh configuration
 
 # Test database and test config.xml have to be created before running tests. This is a one time setup.
-#   1 - create a LorisTest DB and source the default schema (lori/SQL/0000-00-00-schema.sql)
+#   1 - create a LorisTest DB and source the default schemas (SQL/0000-00-*.sql)
 #   2 - create a MySQL user SQLTestUser with password TestPassword.
-#   3 - Modify config.xml file in loris/test/ folder.
-#       Some changes to make in this loris/test/config.xml file:
+#   3 - Modify config.xml file in test/ folder if necessary.
+#       Some changes to verify in this test/config.xml file:
 #       *  Database connection credentials: specify credentials to LorisTest DB which you create in step 1
 #       *  Set sandbox mode to 1: <sandbox>1</sandbox>
 #       *  Set SyncAccounts to false: <SyncAccounts>false</SyncAccounts>
 
-# set environment variable LORIS_DB_CONFIG which is the path to test config.xml file
-export LORIS_DB_CONFIG=$(pwd)/config.xml
-
 #start PHP's built in webserver
-php -S localhost:8000 -t ../htdocs 2>1 > /dev/null &
+php -S localhost:8000 -t ../htdocs 2>&1 > /dev/null &
+php_pid=$!
 
 # Start Selenium and redirect Selenium WebDriver
 # output to /dev/null so that it doesn't flood the
 # screen in the middle of our other tests
-#java -jar /home/kmarasinska/Selenium_server/selenium-server-standalone-2.45.0.jar > /dev/null &
+# java -jar selenium-server-standalone-2.45.0.jar > /dev/null &
 echo "******************************************************************
   REMINDER: Selenium needs to be running to run integration tests
 ******************************************************************";
 
+host="127.0.0.1"
+database="LorisTest"
+username="SQLTestUser"
+password="TestPassword"
+url="http://localhost:8000"
+
+# Custom DB variables specified by optional commandline arguments
+while getopts ":m:h:D:u:p:l:" opt; do
+  case $opt in
+    m) module="$OPTARG"
+    ;;
+    h) host="$OPTARG"
+    ;;
+    D) database="$OPTARG"
+    ;;
+    u) username="$OPTARG"
+    ;;
+    p) password="$OPTARG"
+    ;;
+    l) url="$OPTARG"
+    ;;
+    \?) echo "Invalid option -$OPTARG" >&2
+    ;;
+  esac
+done
+
+# set environment variable LORIS_DB_CONFIG which is the path to test config.xml file
+sed -i \
+    -e "s/%HOSTNAME%/$host/g" \
+    -e "s/%USERNAME%/$username/g" \
+    -e "s/%PASSWORD%/$password/g" \
+    -e "s/%DATABASE%/$database/g" \
+    config.xml
+export LORIS_DB_CONFIG=$(pwd)/config.xml
```

Comment:
Hmm I must be missing something because I'm not sure why the order would really matter so much? Unless we're considering the case that we're trying to do something between the time when the web server gets started up and the time those variables get set.

In any case I'll try to remember to do this while I'm rebasing.